coldfront/core/project/management/commands/projects.py

```diff
@@ -254,3 +358,75 @@ def _validate_create_options(options):
             'compute_resource': compute_resource,
             'pi_users': pi_users,
         }
+
+    @staticmethod
+    def _validate_renew_options(options):
+        """Validate the options provided to the 'renew' subcommand.
+        Raise a subcommand if any are invalid or if they violate
+        business logic, else return a dict of the form:
+            {
+                'project': Project,
+                'requester': User,
+                'pi': User,
+                'allocation_period': AllocationPeriod,
+            }
+        """
+
+        input_username = options['username']
+        requester = None
+        try:
+            requester = User.objects.get(username=input_username)
+        except User.DoesNotExist:
+            raise CommandError(
+                f'User with username "{input_username}" does not exist.')
+
+        project_name = options['name'].lower()
+        if not project_name.startswith("ic_"):
+            raise CommandError(
+                f'The project with name "{project_name}" is not an ICA.'
+            )
+        
+        project = None
+        pi = None
+        try:
+            project = Project.objects.get(name=project_name)
+        except Project.DoesNotExist:
+            raise CommandError(
+                f'A Project with name "{project_name}" does not exist.')
+        
+        if not ProjectUser.objects.filter(project=project, user=requester).exists():
+            raise CommandError(
+                f'Requester is not a user for the project "{project_name}".')
+
+        try:
+            # For ICA projects, there should be only one Principal Investigator
+            pi = ProjectUser.objects.get(
+                project=project, 
+                role=ProjectUserRoleChoice.objects.get(
+                    name='Principal Investigator')
+                ).user
+        except ProjectUser.DoesNotExist:
+            raise CommandError(
+                f'Unable to find the PI for "{project_name}".')
+        
+        input_allocation_period = options['allocation_period']
+        allocation_period = None
+        try:
+            allocation_period = AllocationPeriod.objects.get(name=input_allocation_period)
```

Comment:
Given that we're limiting this to ICAs, we should limit the acceptable periods to those associated with ICAs.

There's logic in the UI form for selecting a period that defines them: https://github.com/ucb-rit/coldfront/blob/master/coldfront/core/project/forms_/new_project_forms/request_forms.py#L95